sdk/armcore/poller.go

```diff
@@ -0,0 +1,1077 @@
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPatch:
+		pt := &pollingTrackerPatch{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPost:
+		pt := &pollingTrackerPost{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	case http.MethodPut:
+		pt := &pollingTrackerPut{pollingTrackerBase: pollingTrackerBase{errorHandler: errorHandler}}
+		// now unmarshal into the tracker
+		err = json.Unmarshal([]byte(token), &pt)
+		if err != nil {
+			return nil, err
+		}
+		return pt, nil
+	default:
+		return nil, fmt.Errorf("unsupported method '%s'", method)
+	}
+}
+
+// Poller defines the methods that will be called internally in the generated code for long-running operations.
+// NOTE: this is only meant for internal use in generated code.
+type Poller interface {
+	// Done signals if the polling operation has reached a terminal state.
+	Done() bool
+	// Poll sends a polling request to the service endpoint and returns the http.Response received from the endpoint or an error.
+	Poll(ctx context.Context, p azcore.Pipeline) (*http.Response, error)
+	// FinalResponse will perform a final GET and return the final http response for the polling operation and unmarshal the content of the payload into the respType interface that is provided.
+	FinalResponse(ctx context.Context, pipeline azcore.Pipeline, respType interface{}) (*http.Response, error)
+	// ResumeToken returns a token string that can be used to resume polling on a poller that has not yet reached a terminal state.
+	ResumeToken() (string, error)
+	// PollUntilDone will handle the entire span of the polling operation until a terminal state is reached, then return the final http response for the polling operation and unmarshal the content of the payload into the respType interface that is provided.
+	PollUntilDone(ctx context.Context, frequency time.Duration, pipeline azcore.Pipeline, respType interface{}) (*http.Response, error)
+}
+
+// Done returns true if there was an error or polling has reached a terminal state.
+func (p *pollingTrackerBase) Done() bool {
+	return p.hasTerminated()
+}
+
+// FinalResponse will perform a final GET and return the final response for the polling operation.
+func (p *pollingTrackerBase) FinalResponse(ctx context.Context, pipeline azcore.Pipeline, respType interface{}) (*http.Response, error) {
+	if !p.Done() {
+		return nil, errors.New("cannot return a final response from a poller in a non-terminal state")
+	}
+	// if respType is nil, this indicates that the request was made from an HTTPPoller
+	if respType == nil {
```

Comment:
FYI this is new and necessary for HTTPPollers

Reply:
In that case can we simply bypass the call to FinalResponse()?

Reply:
No, since we dont expose anything else aside from the Poller interface methods